compat/vcbuild/vcpkg_install.bat

```diff
@@ -33,45 +34,55 @@ REM ================================================================
 
 	SET arch=%1
 	IF NOT DEFINED arch (
-		echo defaulting to 'x64-windows`. Invoke %0 with 'x86-windows', 'x64-windows', or 'arm64-windows'
+		echo defaulting to 'x64-windows`. Invoke %0 with 'x86-windows', 'x64-windows', or 'arm64-windows', append '-static' for static builds. >&2
 		set arch=x64-windows
 	)
 
 	@FOR /F "delims=" %%D IN ("%~dp0") DO @SET cwd=%%~fD
 	cd %cwd%
 
-	dir vcpkg\vcpkg.exe >nul 2>nul && GOTO :install_libraries
+	IF NOT DEFINED VCPKG_ROOT (
+		set VCPKG_ROOT=%cwd%\vcpkg
+	) ELSE (IF NOT EXIST %VCPKG_ROOT%\..\ (
+		echo Invalid VCPKG_ROOT: %VCPKG_ROOT%, not under a valid directory. >&2
+		exit /B 1
+	))
+
+	IF EXIST %VCPKG_ROOT%\vcpkg.exe goto :install_libraries
 
 	git.exe version 2>nul
 	IF ERRORLEVEL 1 (
-	echo "***"
-	echo "Git not found. Please adjust your CMD path or Git install option."
-	echo "***"
-	EXIT /B 1 )
+		echo *** >&2
+		echo Git not found. Please adjust your CMD path or Git install option. >&2
+		echo *** >&2
+		EXIT /B 1
+	)
 
-	echo Fetching vcpkg in %cwd%vcpkg
-	git.exe clone https://github.com/Microsoft/vcpkg vcpkg
+	echo Fetching vcpkg in %VCPKG_ROOT%
+	git.exe clone https://github.com/Microsoft/vcpkg %VCPKG_ROOT%
 	IF ERRORLEVEL 1 ( EXIT /B 1 )
 
-	cd vcpkg
+	cd %VCPKG_ROOT%
 	echo Building vcpkg
 	powershell -exec bypass scripts\bootstrap.ps1
 	IF ERRORLEVEL 1 ( EXIT /B 1 )
 
-	echo Successfully installed %cwd%vcpkg\vcpkg.exe
+	echo Successfully installed %VCPKG_ROOT%\vcpkg.exe
 
 :install_libraries
 
-	echo Installing third-party libraries(%arch%)...
-	FOR %%i IN (zlib expat libiconv openssl libssh2 curl) DO (
-	    cd %cwd%vcpkg
-	    IF NOT EXIST "packages\%%i_%arch%" CALL :sub__install_one %%i
-	    IF ERRORLEVEL 1 ( EXIT /B 1 )
+	echo Installing third-party libraries(%arch%), this may take a while...
+	FOR %%i IN (zlib expat libiconv openssl libssh2 curl gettext) DO (
```

Comment:
I am fairly certain that we do _not_ want to include `gettext` here by default: As discussed, this adds quite a chunk of time to the overall build time. This is especially unwanted because the CMake build is _not_ the way official Git for Windows is built, i.e. it definitely targets contributors who are much more likely to be happy to build without translations if that means that they can work on the Git source code quicker.

Reply:
How about making `NO_GETTEXT default to `TRUE` then?

Reply:
I mean as the default.

Reply:
I would rather detect the absence of `msgfmt.exe` or `libintl` and default to `NO_GETTEXT` in those cases.